restricted_site_access.php

```diff
@@ -350,6 +385,13 @@ public static function restrict_access( $wp ) {
 				if ( ( $current_url_path === $redirect_url_without_scheme ) || ( $redirect_url_without_scheme === $current_url_without_scheme ) ) {
 					return;
 				}
+
+				$redirection_url_host = trailingslashit( wp_parse_url( $results['url'], PHP_URL_HOST ) );
+				$current_url_host     = trailingslashit( wp_parse_url( home_url( $wp->request ), PHP_URL_HOST ) );
+
+				if ( $redirection_url_host === $current_url_host || '/' === $redirection_url_host ) {
+					setcookie( 'wp-rsa_redirect', self::generate_redirection_cookie( $results['url'] ) );
+				}
```

Comment:
```suggestion
				$redirection_url_host = trailingslashit( wp_parse_url( $results['url'], PHP_URL_HOST ) );
				$current_url_host     = trailingslashit( wp_parse_url( home_url( $wp->request ), PHP_URL_HOST ) );
				$cookie_path          = wp_parse_url( home_url( '/' ), PHP_URL_PATH );

				if ( $redirection_url_host === $current_url_host || '/' === $redirection_url_host ) {
					setcookie( 'wp-rsa_redirect', self::generate_redirection_cookie( $results['url'] ), 0, $cookie_path );
				}
```

Ensures the cookie is available site wide. In setcookie, the path defaults to the current path so visiting `example.com/john` could set a cookie that is unavailable if the redirect is `example.com/paul`